### DISCUSSION TITLE
大伙有没有思路用Taro实现直接导出H5原生工程可用的组件实现？

### DISCUSSION THREAD

**Discussion by gMorning-Wp** [NervJS/taro#11925] [Q & A]
我遇到的问题是这样的，我们现在有一个creat-react-app生成的H5 React原生工程用于H5开发工作，有一个H5自身的组件库服务于前面的H5工程；又有一个 Taro-cli生成的用于小程序开发的工程，我们业务提出一个需求，需要完成俩份代码的编写工作。我们希望直接用taro来实现H5和小程序的一套代码的编写生成页面（这部分已经利用taro的能力实现了）。**但还有一个点：就是希望用taro的一套代码实现导出H5 React原生写法的组件能应用在creat-react-app工程中。对于这个点大家有没有好的思路和实现呢**

**Comment by Chen-jj** [comment]
现在是支持的，因为 Taro H5 也是普通的 H5 项目，他通过一些编译配置去使用小程序规范的组件库 @tarojs/components 和 API 库。文档里有一个 Taro 组件库在 Storybook 中使用的例子可以参考下。当然如果能封装成插件开放给社区就更好了。https://docs.taro.zone/docs/external-libraries#storybook

**Reply by gMorning-Wp** [reply]
万分感谢你的回复！是否在taro-uilib-react工程中用rollup去实现一个直接转成H5组件的插件更为合理？

**Reply by biorz** [reply]
这个我写过， 不过单靠 rollup 是不行的， 还是需要改动到项目的 webpack 配置。

**Reply by zhouzg5533** [reply]
> 这个我写过， 不过单靠 rollup 是不行的， 还是需要改动到项目的 webpack 配置。
@biorz 请问有实现的demo吗？可否提供学习一下，感谢

**Reply by biorz** [reply]
> > 这个我写过， 不过单靠 rollup 是不行的， 还是需要改动到项目的 webpack 配置。
> > @biorz 请问有实现的demo吗？可否提供学习一下，感谢

目前是发在内网的，等我抽出来放在 github 上。@zhouzg5533

**Reply by zhouzg5533** [reply]
> > > 这个我写过， 不过单靠 rollup 是不行的， 还是需要改动到项目的 webpack 配置。
> > > @biorz 请问有实现的demo吗？可否提供学习一下，感谢
> 
> 目前是发在内网的，等我抽出来放在 github 上。@zhouzg5533

@biorz 万分感谢~~~能否简单提供一些思路？因为近期打算做这个事情，如果有现成的demo或者资料，麻烦发我邮箱可以嘛？zhouzegao5533@163.com。

**Reply by Wuhu-dsm** [reply]
> > > > 这个我写过， 不过单靠 rollup 是不行的， 还是需要改动到项目的 webpack 配置。
> > > > @biorz 请问有实现的demo吗？可否提供学习一下，感谢
> > 
> > 
> > 目前是发在内网的，等我抽出来放在 github 上。@zhouzg5533
> 
> @biorz 万分感谢~~~能否简单提供一些思路？因为近期打算做这个事情，如果有现成的demo或者资料，麻烦发我邮箱可以嘛？[zhouzegao5533@163.com](mailto:zhouzegao5533@163.com)。

大佬，有实现吗？我也有这个需求

**Comment by paopaoma** [comment]
实现了嘛老哥 我也有这个问题 想把taro的组件 打包成直接可用的 h5 react组件

**Reply by Wuhu-dsm** [reply]
实现了吗，老哥！

**Reply by honlyHuang** [reply]
参考：https://github.com/NervJS/taro/tree/main/examples/blended-taro-component
